addons/mail/static/src/components/attachment_viewer/attachment_viewer.js

```diff
@@ -137,7 +137,10 @@ export class AttachmentViewer extends Component {
      */
     _download() {
         const id = this.attachmentViewer.attachment.id;
-        this.env.services.navigate(`/web/content/ir.attachment/${id}/datas`, { download: true });
+        let downloadLink = document.createElement('a');
+        downloadLink.setAttribute('href', `/web/content/ir.attachment/${id}/datas`);
```

Comment:
This url is wrong for guest, but out of scope, it will be fixed in task-2664842.